### PR TITLE
refactor(backend): revise authentication for endpoints

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

[Ticket](https://www.notion.so/opengov/Revise-cookie-API-authentication-1fcff2fca43d425a9e735350f5734e0e)

## Solution

- Make auth middleware modular
- Restrict non-API-exposed endpoints to be cookie-only
- Remove non-API-exposed from our published swagger doc

## Deployment Checklist

_Any tasks that need to be done on the environment before releasing of this PR (e.g. Setting environment variables, running migrations, etc). If there's no task to be done, put "N/A"_

- N/A

### CODE REVIEW

File: backend/src/core/middlewares/auth.middleware.ts [hunk 1]
```diff
@@ -110,48 +115,70 @@ export const InitAuthMiddleware = (authService: AuthService) => {
...
-    try {
-      if (authService.checkCookie(req)) {
-        logger.info({
-          message: 'User authenticated by cookie',
-          action: 'isCookieOrApiKeyAuthenticated',
-          user: req.session?.user,
-        })
-        return next()
+  type AuthenticatorType = (req: Request) => Promise<boolean>
+  const authenticators: Record<AuthType, AuthenticatorType> = {
```
Comment (richard-yak) [inline backend/src/core/middlewares/auth.middleware.ts:119]: wow this is some insane use of generics to do what i would've written as a basic switch case to handle. nice.

**Review by richard-yak — APPROVED** [review]
i've learnt

**Comment by tan-zx** [comment]
oh btw since you're making these changes, could you also add in the `security: bearerAuth[]` as shown below?

image

it'll help make the Swagger docs look nicer by clearly distinguishing public routes and API-authenticated routes with the little lock sign as shown below

image

**Review by richard-yak — APPROVED** [review]
2nd review np

**Pull request merged**